### PR TITLE
Avoid -Wall msg in V0-jet code and add protections

### PR DESCRIPTION
**Pull request by xcheung**
- Avoid potential override methods in parent class in derived class
- Add const protection for methods used to return V0 kinematics and that used for V0 selection
- V0 selection using Polymorphism of parent class

### CODE REVIEW

**Comment by xcheung** [comment]
Dear approvers: The update of the code has been tested at local. It gives the consistent results as the previous version. Neither crash nor compilation warning/error was found.

File: PWGJE/FlavourJetTasks/AliPicoV0.h [hunk 1]
```diff
@@ -31,21 +31,29 @@ class AliPicoV0 : public TObject {
   virtual ~AliPicoV0();
 //=============================================================================
 
-  TVector3 KinePos() const { return  fP3Pos; }
-  TVector3 KineNeg() const { return  fP3Neg; }
-  TVector3 KineRD()  const { return (fP3Pos + fP3Neg); }
+  const TVector3 KinePos() const { return  fP3Pos; }
+  const TVector3 KineNeg() const { return  fP3Neg; }
```
Comment (mfasDa) [inline PWGJE/FlavourJetTasks/AliPicoV0.h]: Could return const TVector3 & in order to avoid unnecessary copying.

**Comment by xcheung** [comment]
@mfasDa Thanks for you suggestion. The code is updated.

File: PWGJE/FlavourJetTasks/AliPicoV0.h (hunk 1 above)
Comment (xcheung) [reply PWGJE/FlavourJetTasks/AliPicoV0.h]: Thanks for your suggestion. The code is updated.

**Comment by mfasDa** [comment]
+test

**Comment by mfasDa** [comment]
+1

**Pull request merged**